README.md

```diff
@@ -50,17 +50,74 @@ engine.push(function(req, res, next, end){
 })
 ```
 
-RpcEngines can be nested by converting them to middleware `asMiddleware(engine)`
+RpcEngines can be nested by converting them to middleware using `asMiddleware(engine)`:
 
 ```js
-const asMiddleware = require('json-rpc-engine/lib/asMiddleware')
+const asMiddleware = require('json-rpc-engine/src/asMiddleware')
 
 let engine = new RpcEngine()
 let subengine = new RpcEngine()
 engine.push(asMiddleware(subengine))
 ```
 
-### gotchas
+### `async` Middleware
+
+If you require your middleware function to be `async`, use `createAsyncMiddleware`:
+
+```js
+const createAsyncMiddleware = require('json-rpc-engine/src/createAsyncMiddleware')
+
+let engine = new RpcEngine()
+engine.push(createAsyncMiddleware(async (req, res, next) => {
+  req.result = 42
+  next()
+}))
+```
+
+`async` middleware do not take an `end` callback.
+Instead, the request ends if the middleware returns without calling `next()`:
+
+```js
+engine.push(createAsyncMiddleware(async (req, res, next) => {
+  req.result = 42
+  /* The request will end when this returns */
+}))
+```
+
+The `next` callback of `async` middleware also don't take return handlers.
+Instead, you can `await next()`.
+When the execution of the middleware resumes, you can work with the response.
```

Comment:
Maybe this could be updated to clarify that working with the response is still possible before this point as well? Just that after `next()`, you're given the chance to work with the response again _after_ it has been processed by all other middleware below in the stack. Not sure how to word that 🤔 

Unless this is wrong? That's how I thought it worked.

Reply:
Maybe this is fine:
```suggestion
When the execution of the middleware resumes, you can work with the response again.
```